snuba/migrations/migration.py

```diff
@@ -87,3 +91,12 @@ def forwards(self, context: Context) -> None:
             op.execute()
         logger.info(f"Finished: {migration_id}")
         update_status(Status.COMPLETED)
+
+    def backwards(self, context: Context) -> None:
+        migration_id, logger, update_status = context
+        logger.info(f"Reversing migration: {migration_id}")
+        update_status(Status.IN_PROGRESS)
+        for op in self.backwards_local():
+            op.execute()
+        logger.info(f"Finished reversing: {migration_id}")
+        update_status(Status.NOT_STARTED)
```

Comment:
Would there be any value in keeping track of a "reversed" state to signify that we actually tried and failed ? I think it may be useful for troubleshooting.

Reply:
I am not sure about this. To me the reversed state is exactly the same as not started and doesn't actually represent a different underlying state.

It sounds to me like what you are wanting to have is more like a history of all status changes that have been written to the database. After all, migrationscan potentially be run and reversed many times.

We can already infer whether something was reversed by whether the NOT STARTED status is written into the migrations table, and what the version number is.

Reply:
It is about troubleshooting to see understand the exact state of the system after an issue. A full history is different. 
I don't think relying on "not started" + version number is the best option. While it is true that in the current design that would work, it is more of a coincidence that we can extract the same information from that combination of states, identifying a state with a state name is generally preferred to implying the state from a combination of independent conditions. This is because it would be easy to make changes in the implementation that would drop such implicit assumption (that we can use not started + version to tell whether we reverted) without noticing the consequences.

Of course if the complexity of adding a new state was very high, this would not apply, but it does not seem the case here.

Reply:
oh right ..... clickhouse enums. 

ok, nevermind.

Reply:
Wait, what do you mean by clickhouse enums? I think adding it in will be easy. Just want to be sure we are not actually looking for full history or something else here.